### PR TITLE
Add support for STM32F4DISCOVERY Discovery kit and fix support for STM32F407XG

### PR DESCRIPTION
**Pull request by superna9999**
This PR fixes support for the STM32F407XG SoC and support for the STM32F4DISCOVERY Discovery kit board.

Documentation for the STM32F4DISCOVERY Discovery kit board is added based on the Nucleo F411RE format.

### CODE REVIEW

**Review by erwango — CHANGES_REQUESTED** [review]
Please rename board as stm32f4_disco for consistency and we could easily identify _disco boards.
Agreed this is not in line with ST naming on website, but I'm afraid st naming convention is not clear enough

**Comment by superna9999** [comment]
@erwango ok, But should I leave all the STM32F4DISCOVERY naming in the doc ? and only reference stm32f4_disco in the zephyr related text ?

**Comment by erwango** [comment]
Indeed, let it as is in doc.

File: drivers/pinmux/stm32/pinmux_board_stm32f4discovery.c [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+static const struct pin_config pinconf[] = {
+#ifdef CONFIG_UART_STM32_PORT_1
+	{STM32_PIN_PB6, STM32F4_PINMUX_FUNC_PB6_USART1_TX},
+	{STM32_PIN_PB7, STM32F4_PINMUX_FUNC_PB7_USART1_RX},
+#endif	/* CONFIG_UART_STM32_PORT_1 */
+#ifdef CONFIG_UART_STM32_PORT_2
+	{STM32_PIN_PA2, STM32F4_PINMUX_FUNC_PA2_USART2_TX},
+	{STM32_PIN_PA3, STM32F4_PINMUX_FUNC_PA3_USART2_RX},
+#endif	/* CONFIG_UART_STM32_PORT_2 */
+};
```
Comment (superna9999) [inline drivers/pinmux/stm32/pinmux_board_stm32f4discovery.c]: no idea on which pin to set PWM, so I'll remove this config

File: drivers/pinmux/stm32/pinmux_board_stm32f4discovery.c [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <pinmux.h>
+#include <sys_io.h>
+#include "pinmux/pinmux.h"
+
+#include "pinmux_stm32.h"
+
+/* pin assignments for NUCLEO-F401RE board */
```
Comment (superna9999) [inline drivers/pinmux/stm32/pinmux_board_stm32f4discovery.c]: Ok

File: boards/arm/stm32f4discovery/stm32f4discovery_defconfig [hunk 3]
```diff
@@ -0,0 +1,47 @@
+CONFIG_ARM=y
+CONFIG_BOARD_STM32F4DISCOVERY=y
+CONFIG_SOC_SERIES_STM32F4X=y
+CONFIG_SOC_STM32F407XG=y
+# 84MHz system clock
```
Comment (superna9999) [inline boards/arm/stm32f4discovery/stm32f4discovery_defconfig]: Ok

**Review by fallrisk — CHANGES_REQUESTED** [review]
You don't have a fixup file for DTS. Currently your board won't compile because you don't have NUM_IRQ_PRIO_BITS defined. Look at any of the fixup files for an example.

**Comment by superna9999** [comment]
Hi @fallrisk, It was dropped in the board renaming...

**Review by fallrisk — APPROVED** [review]
Yeah, that happens. I get a bunch of stuff working in one branch and then I forget things when I split it to make the PR.

**Review by dbkinder — APPROVED** [review]
LGTM

**Comment by superna9999** [comment]
I'll rebase this PR once #126 is merged.

**Comment by galak** [comment]
#126 is now merged

**Pull request merged**